xgi/classes/reportviews.py

```diff
@@ -688,3 +692,48 @@ def singletons(self):
 
         """
         return self.filterby("size", 1)
+
+    def maximal(self):
+        """Returns the maximal edges as an EdgeView
+
+        Returns
+        -------
+        EdgeView
+            The maximal edges
+
+        Notes
+        -----
+        When there are maximal edges that are also multi-edges,
+        `maximal()` returns all of these multi-edges rather than
+        choosing one of them to return. There are methods for
+        eliminating these duplicates by running `H.cleanup()`
+        or `H.remove_edges_from(H.edges.duplicates())`
+
+        Example
+        -------
+
+        >>> import xgi
+        >>> H = xgi.Hypergraph([{1, 2, 3},{1, 2}, {2, 3}, {2}, {2}, {3, 4}, {1, 2, 3}])
+        >>> H.edges.maximal()
+        EdgeView((0, 5, 6))
+        >>> H.edges.maximal().members()
+        [{1, 2, 3}, {3, 4}, {1, 2, 3}]
+        """
+        edges = self._id_dict
+        nodes = self._bi_id_dict
+        max_edges = set()
+
+        hashes = defaultdict(list)
+        for idx, members in edges.items():
+            hashes[frozenset(members)].append(idx)
+
+        _intersection = lambda x, y: x & y
+
+        for i, e in edges.items():
+            if i not in max_edges:
+                in_common = reduce(_intersection, [nodes[n] for n in e])
```

Comment:
More memory efficient, and I don't think it would have much impact in running time:
```suggestion
                in_common = reduce(_intersection, (nodes[n] for n in e))
```